apps/src/applab/applab.js

```diff
@@ -1196,20 +1178,18 @@ Applab.execute = function () {
     }
   }
 
-  if (isMakerEnabled(getStore().getState())) {
-    connectToMakerBoard()
-        .then(board => {
-          board.installOnInterpreter(codegen, Applab.JSInterpreter);
-          makerCommands.injectBoardController(board);
-          board.once('disconnect', () => studioApp.resetButtonClick());
-          makerBoard = board;
-        })
-        .catch(error => studioApp.displayPlayspaceAlert('error',
-            <div>{`Board connection error: ${error}`}</div>))
-        .then(Applab.beginVisualizationRun);
-  } else {
-    Applab.beginVisualizationRun();
-  }
+    makerToolkit.connect({
+      interpreter: Applab.JSInterpreter,
+      onDisconnect: () => studioApp.resetButtonClick(),
+    })
+        .then(Applab.beginVisualizationRun)
+        .catch(error => {
+          // Don't just throw any error away, but squelch errors that we already
+          // handle gracefully (like early disconnect or a missing board).
+          if (!(error instanceof makerToolkit.MakerError)) {
+            Applab.log(error);
```

Comment:
rethrow? I'd rather have the exception bubble all the way up if something unexpected happened, but I could be convinced otherwise.